ArduPlane/Attitude.cpp

```diff
@@ -454,7 +454,7 @@ void Plane::calc_throttle()
     int32_t commanded_throttle = SpdHgt_Controller->get_throttle_demand();
 
     // Received an external msg that guides throttle in the last 3 seconds?
-    if ((control_mode == &mode_guided || control_mode == &mode_avoidADSB) &&
+    if (control_mode->is_guided() &&
```

Comment:
This actually looks like a preexsisting bug. If you are in ADS-B avoidance mode why would we want to use the externally forced throttle? This was provided by a companion computer/GCS of some form as part of the attitude control messages, ADS-B avoidance starts breaking down if we use the GCS's overrides instead of allowing the system to calculate all these dynamically.

Reply:
yes, I agree